frontend/src/components/my-courses/my-courses.tsx

```diff
@@ -0,0 +1,97 @@
+import { DataStatus, PaginationDefaultValue } from 'common/enums/enums';
+import { CourseUpdateMentoringDto, FC } from 'common/types/types';
+import { CoursesList, Spinner, Table } from 'components/common/common';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+  useMemo,
+  usePagination,
+} from 'hooks/hooks';
+import { Column } from 'react-table';
+import { myCoursesActions } from 'store/actions';
+
+import { CoursesMentoringTableRow } from './common/types/types';
+import {
+  getCoursesMentoringColumns,
+  getCoursesMentoringRows,
+} from './helpers/helpers';
+import styles from './styles.module.scss';
+
+const MyCourses: FC = () => {
+  const { page, handlePageChange } = usePagination({
+    queryName: 'myCoursesPage',
+  });
+
+  const dispatch = useAppDispatch();
+  const {
+    coursesStudying,
+    coursesMentoring,
+    totalCoursesMentoring,
+    dataStatus,
+  } = useAppSelector((state) => ({
+    coursesStudying: state.myCourses.coursesStudying,
+    coursesMentoring: state.myCourses.coursesMentoring,
+    dataStatus: state.myCourses.dataStatus,
+    totalCoursesMentoring: state.myCourses.totalCoursesMentoring,
+  }));
+
+  useEffect(() => {
+    dispatch(myCoursesActions.getCoursesStudying());
+    dispatch(
+      myCoursesActions.getCoursesMentoring({
+        page,
+        count: PaginationDefaultValue.DEFAULT_COUNT,
+      }),
+    );
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(
+      myCoursesActions.getCoursesMentoring({
+        page,
+        count: PaginationDefaultValue.DEFAULT_COUNT,
+      }),
+    );
+  }, [page]);
+
+  const handleEdit = (course: CourseUpdateMentoringDto): void => {
+    dispatch(myCoursesActions.updateCoursesMentoring(course)).then(() => {
+      dispatch(
+        myCoursesActions.getCoursesMentoring({
+          page,
+          count: PaginationDefaultValue.DEFAULT_COUNT,
+        }),
+      );
+    });
```

Comment:
I think there's no need to request the courses again, after updating the number of students, the value of students number can be changed in reducer case for `updateCoursesMentoring`